msvs-promote-path

```diff
@@ -28,36 +28,33 @@
 # ################################################################################################ #
 
 # Ensure that the Microsoft Linker isn't being messed up by /usr/bin/link
-if [ "$(link --version 2>/dev/null | sed -ne "1s/Microsoft (R) Incremental Linker//p")" != "" ] ; then
+# We expect the Microsoft Linker to be in the same directory than the C compiler
+
+if ! clpath="$(command -v cl)" ; then
+  echo "The Microsoft C compiler was not found in any of the PATH entries!">&2
+  exit 1
+fi
+clpath="${clpath%/*}"
+
+if ! linkpath="$(command -v link)" ; then
+  echo "The Microsoft Linker was not found in any of the PATH entries!">&2
+  exit 1
+fi
+
+if [ "${linkpath%/*}" = "$clpath" ]; then
   echo "link already refers to the Microsoft Linker">&2
   exit 0
 fi
 
+NEWPATH="$clpath"
 IFS=:
-T=
-FOUND=0
-FIRST=1
 for i in $PATH
 do
-  if [ $FIRST -eq 1 ] ; then
-    T="$i"
-    FIRST=0
-  else
-    if [ $FOUND -eq 0 ] && [ -x "$i/link" ] && [ "$("$i/link" --version 2>/dev/null | sed -ne "1s/Microsoft (R) Incremental Linker//p")" != "" ] ; then
-      FOUND=1
-      T="$i:$T"
-      PROM=$i
-    else
-      T="$T:$i"
-    fi
+  if [ "$i" '!=' "$clpath" ]; then
+    NEWPATH="$NEWPATH:$i"
   fi
 done
 unset IFS
 
-if [ $FOUND -eq 0 ] ; then
-  echo "The Microsoft Linker was not found in any of the PATH entries!">&2
-  exit 1
-else
-  echo "$PROM moved to the front of \$PATH">&2
-  echo "export PATH='$(echo "$T" | sed -e "s/'/'\"'\"'/g")'"
-fi
+echo "$clpath moved to the front of \$PATH">&2
+echo "export PATH=\"$NEWPATH\""
```

Comment:
Unless I'm missing something, this can't be done in double-quotes without escaping (or `/cygdrive/c/Devel/$(rm -rf .)` becomes a worry!). However, we can still do better than my rubbish use of `sed`, I think, with:
```suggestion
echo "export PATH='"${NEWPATH//\'/\'\"\'\"\'}"'"
```

Reply:
We could also use `$(printf %q "$NEWPATH")` if `printf` is indeed a built-in in all the versions of bash we’d like to target?